### ISSUE TITLE
Compile Error with VSCode

### ISSUE THREAD

**Issue by Artourou** [angular/flex-layout#1016]
# Bug Report

#### What is the expected behavior?
I followed the steps in https://github.com/angular/flex-layout/wiki/Developer-Setup but when I execute this command : npm run demo:serve, the VSCode editor display an error :

Error : spawn npm ENOENT
at Process.childProcess._handle.onexit (internal/child_process.js:232:19)
at onErrorNT (internal/child_process.js:407:16)
at process._ticketCallback (internal/process/next_tick.js:63:19)

#### What is the current behavior?
I wanted to run the demo source code but I got this error  
![error - spawn npm enoent](https://user-images.githubusercontent.com/47301070/52179744-40821280-27de-11e9-895a-256ebc55ae6a.png)

#### What are the steps to reproduce?
To reproduce this error, just follow the steps provided by Flex-Layout team (https://github.com/angular/flex-layout/wiki/Developer-Setup )  

1. Make sure you have node installed with a version at least 8.9.1
2. Run npm install -g gulp to install gulp
3. Fork the angular/flex-layout repo
4. Clone your fork.
5. run npm run lib:build
6. run npm run demo:serve

Providing a StackBlitz (or similar) is the *best* way to get the team to see your issue.

*  [StackBlitz starter](https://goo.gl/y38GUZ) using latest `npm` nightly release: 

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
   - OS Windows 10 X64
   - VScode 1.30.2
   - node V10.12.0
   - npm V 6.5.0
   - TypeScript 3.2
  - Angular CLI 7.2.1

#### Is there anything else we should know?

I googled and found that we can add a launch.json file inside .vscode to allow debug using npm.cmd but it didn't work.

**Comment by CaerusKaru** [comment]
I just ran it locally and it works fine on my end with a fresh install. This looks like it could potentially be a Windows problem, so I'll try and get someone with Windows to confirm.

**Comment by johnpankowicz** [comment]
I have the same error.  This has nothing to do with VSCode. VSCode is just running the npm commands in its terminal window. @Artourou : If you close VSCode and run these commands in a separate window, you should have the same results.
I cloned the project, ran "npm install", ran that fixes suggested by "npm audit" and then tried to run "npm run lib:build". I immediately get the same error output that you show.
It appears that this may be a gulp setup issue: [gulp/issues/2349](https://github.com/gulpjs/gulp/issues/2349)
 Below is the debug log file contents:

```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'lib:build' ]
2 info using npm@6.9.0
3 info using node@v10.16.1
4 verbose run-script [ 'prelib:build', 'lib:build', 'postlib:build' ]
5 info lifecycle flex-layout-srcs@8.0.0-beta.27~prelib:build: flex-layout-srcs@8.0.0-beta.27
6 info lifecycle flex-layout-srcs@8.0.0-beta.27~lib:build: flex-layout-srcs@8.0.0-beta.27
7 verbose lifecycle flex-layout-srcs@8.0.0-beta.27~lib:build: unsafe-perm in lifecycle true
8 verbose lifecycle flex-layout-srcs@8.0.0-beta.27~lib:build: PATH: C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;D:\SOURCECODE\_GITHUB\angular\flex-layout\node_modules\.bin;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\bin\Roslyn;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Team Tools\Performance Tools;C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\vs2019\;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\IDE\CommonExtensions\Microsoft\FSharp\;C:\Program Files (x86)\Windows Kits\10\bin\10.0.17763.0\x86;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\\MSBuild\Current\Bin;C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\Tools\;C:\Program Files (x86)\Git\bin;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files\ffmpeg\bin;C:\Program Files\dotnet\;C:\Program Files (x86)\GtkSharp\2.12\bin;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\ManagementStudio\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\Yarn\bin\;C:\Program Files (x86)\dotnet\;C:\ProgramData\chocolatey\bin;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\Users\johnp\AppData\Local\Microsoft\WindowsApps;C:\Program Files (x86)\Microsoft VS Code\bin;C:\Users\johnp\AppData\Local\Google\Cloud SDK\google-cloud-sdk\bin;C:\Users\johnp\AppData\Local\Yarn\bin;C:\Users\johnp\.dotnet\tools;D:\_MyTools\_MyExe;D:\_MyTools\_MyScripts;C:\Users\johnp\AppData\Local\Programs\Python\Python37-32;C:\Users\johnp\AppData\Local\Programs\Python\Python37-32\Scripts;C:\Program Files (x86)\sox-14-4-2;C:\Users\johnp\AppData\Local\Programs\Fiddler;C:\Users\johnp\AppData\Roaming\npm;C:\Users\johnp\AppData\Local\Programs\Microsoft VS Code\bin;C:\Program Files (x86)\DeltaCopy;C:\Users\johnp\AppData\Local\Microsoft\WindowsApps;C:\Program Files (x86)\sqlite;
9 verbose lifecycle flex-layout-srcs@8.0.0-beta.27~lib:build: CWD: D:\SOURCECODE\_GITHUB\angular\flex-layout
10 silly lifecycle flex-layout-srcs@8.0.0-beta.27~lib:build: Args: [ '/d /s /c', 'gulp :publish:build-releases' ]
11 silly lifecycle flex-layout-srcs@8.0.0-beta.27~lib:build: Returned: code: 1  signal: null
12 info lifecycle flex-layout-srcs@8.0.0-beta.27~lib:build: Failed to exec lib:build script
13 verbose stack Error: flex-layout-srcs@8.0.0-beta.27 lib:build: `gulp :publish:build-releases`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:198:13)
13 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:198:13)
13 verbose stack     at maybeClose (internal/child_process.js:982:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
14 verbose pkgid flex-layout-srcs@8.0.0-beta.27
15 verbose cwd D:\SOURCECODE\_GITHUB\angular\flex-layout
16 verbose Windows_NT 10.0.18362
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "lib:build"
18 verbose node v10.16.1
19 verbose npm  v6.9.0
20 error code ELIFECYCLE
21 error errno 1
22 error flex-layout-srcs@8.0.0-beta.27 lib:build: `gulp :publish:build-releases`
22 error Exit status 1
23 error Failed at the flex-layout-srcs@8.0.0-beta.27 lib:build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by johnpankowicz** [comment]
I just found the cause of my problem. I should not have fixed the "npm audit" vulnerabilities. One of the suggested fixes was to run "npm install --save-dev gulp@4.0.2". It turns out that Gulp 4 changes the way tasks are defined and most likely is not compatible with some of the defined tasks in angular/flex-layout. See: [SO question 51098749](https://stackoverflow.com/questions/51098749/everytime-i-run-gulp-anything-i-get-a-assertion-error-task-function-must-be)
I removed node_modules and package-lock.json. I reset the contents of package.json back to what is in the repo.
After running "npm install", I could run "npm run lib:build" with no errors.
